recipes/pciutils/all/conanfile.py

```diff
@@ -1,76 +1,128 @@
 import os
-from conans import ConanFile, tools, AutoToolsBuildEnvironment
-from conans.errors import ConanInvalidConfiguration
+import shutil
+
+from conan import ConanFile
+from conan.errors import ConanInvalidConfiguration
+from conan.tools.apple import XCRun
+from conan.tools.env import VirtualBuildEnv
+from conan.tools.files import chdir, copy, get, rmdir
+from conan.tools.gnu import Autotools, AutotoolsToolchain, AutotoolsDeps
+from conan.tools.layout import basic_layout
+
+required_conan_version = ">=1.53.0"
 
 
 class PciUtilsConan(ConanFile):
     name = "pciutils"
-    license = "BSD-3-Clause"
     description = "The PCI Utilities package contains a library for portable access to PCI bus"
-    topics = ("pci", "pci-bus", "hardware", "local-bus")
-    homepage = "https://github.com/pciutils/pciutils"
+    license = "BSD-3-Clause"
     url = "https://github.com/conan-io/conan-center-index"
-    settings = "os", "compiler", "build_type", "arch"
-    options = {"shared": [True, False], "fPIC": [True, False], "with_zlib": [True, False], "with_udev": [True, False]}
-    default_options = {"shared": False, "fPIC": True, "with_zlib": True, "with_udev": False}
+    homepage = "https://github.com/pciutils/pciutils"
+    topics = ("pci", "pci-bus", "hardware", "local-bus")
 
-    @property
-    def _source_subfolder(self):
-        return "source_subfolder"
+    package_type = "library"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "with_zlib": [True, False],
+        "with_udev": [True, False],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "with_zlib": True,
+        "with_udev": True,
+    }
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
 
     def configure(self):
-        if self.settings.os != "Linux":
-            raise ConanInvalidConfiguration("Platform {} is currently not supported by this recipe".format(self.settings.os))
+        if self.settings.os not in ["Linux", "FreeBSD"]:
```

Comment:
Shouldn't this be in the `validate()` method?

Reply:
Absolutely. Thanks!